library/psa_crypto_core.h

```diff
@@ -50,13 +50,28 @@ typedef struct {
      * Number of locks on the key slot held by the library.
      *
      * This counter is incremented by one each time a library function
-     * retrieves through one of the dedicated internal API a pointer to the
-     * key slot.
+     * calls the psa_lock_key_slot() API, library functions must do this
+     * before they read the current content of the slot for an operation.
+     *
+     * This counter is set to SIZE_MAX by a call to
+     * psa_lock_key_slot_for_writing(), library functions must do this
+     * before they write to any field in the slot other than lock_count.
```

Comment:
"before they write to any field in the slot other than lock_count"

Actually, it sounds like library functions must NEVER directly write to `lock_count` - in which case, why not state that explicitly at the top of this documentation block, then document what functions must be called to read/write lock the slot, and maybe even have the values used for this documented privately within those functions?

Reply:
I have re-written this bit of the documentation as suggested, I think it reads more clearly now.